internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -160,10 +160,10 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
         case _ => newOne()
       }
     }
-    private def addClassDependency(deps: HashSet[ClassDependency], fromClass: Symbol, dep: Symbol): Unit = {
+    private def addClassDependency(deps: HashSet[ClassDependency], fromClass: Symbol, dep: Symbol): Unit = if (dep != NoSymbol) {
       assert(fromClass.isClass, Feedback.expectedClassSymbol(fromClass))
       val depClass = enclOrModuleClass(dep)
-      if (fromClass.associatedFile != depClass.associatedFile) {
+      if (fromClass.associatedFile != depClass.associatedFile && !depClass.isRefinementClass) {
```

Comment:
@jvican Refinement classes do not have byte code representation and it fails later on (we still depends on parents of refinement)